addon/components/nav-category.js

```diff
@@ -9,36 +10,40 @@ const {
   run: {
     later
   },
-  computed
+  get
 } = Ember
 
-export default Component.extend({
+export default Component.extend(PropTypesMixin, {
   frostNavigation: service(),
   classNames: ['nav-category'],
   classNameBindings: ['active'],
-  active: computed('frostNavigation._activeCategory', function () {
-    return this.name === this.get('frostNavigation._activeCategory')
-  }),
+  @computed('frostNavigation._activeCategory')
+  active (category) {
+    return get(this, 'name') === category
+  },
   layout,
   propTypes: {
     icon: PropTypes.string,
     name: PropTypes.string.isRequired,
     pack: PropTypes.string
   },
   click () {
-    let navService = this.get('frostNavigation')
+    let navService = get(this, 'frostNavigation')
     if (!navService) return
-    let activeCategory = navService.get('_activeCategory')
-    let name = this.get('name')
+    document.body.scrollTop = 0 // fix for liquid-fire modal strange animation
+    let activeCategory = get(navService, '_activeCategory')
+    let name = get(this, 'name')
     if (name === activeCategory) {
       navService.dismiss()
-    } else if (typeof activeCategory === 'string') {
-      navService.dismiss()
-      later(() => {
-        navService.set('_activeCategory', name)
-      }, 100)
     } else {
-      navService.set('_activeCategory', name)
+      if (typeof activeCategory === 'string') {
+        navService.dismiss()
+        later(() => {
+          navService.set('_activeCategory', name)
```

Comment:
You can destructure the `set` as well.

`set(navService, '_activeCategory', name)` 

Remember to define it at the top.